src/queries/SyncPullQuery.ts

```diff
@@ -0,0 +1,26 @@
+export interface SyncPullQuery {
```

Comment:
It is better to abstract some general definition. Or you can go to GitHub and look for it.

Reply:
In the current design, query class and method are strongly related, the attributes in the query class are defined according to the return type of the API, but some unused attributes are removed. So it's hard to abstract a general definition.

Maybe we can directly use the type definition in octokit, what do you think?

Reply:
Yes, it is best if we can use the existing definitions.

Reply:
> Yes, it is best if we can use the existing definitions.

But there is a problem that there are too many useless properties in the existing definitions, which means that the require properties need to be passed in when calling, especially when testing. 

This is not as good as passing in a condensed query object to better explain those properties are necessary.

Reply:
OK, you can use your own quires.